### PR TITLE
Ensure receiver exists before delivering message

### PR DESCRIPTION
**Pull request by mtokioka**
Add checking existence of receiver because calling publish & unsubscribe at the same time might cause receiving message for deleted receiver.

### CODE REVIEW

File: lib/gnat.ex [hunk 1]
```diff
@@ -249,8 +249,11 @@ defmodule Gnat do
   end
 
   defp process_message({:msg, topic, sid, reply_to, body}, state) do
-    send state.receivers[sid].recipient, {:msg, %{topic: topic, body: body, reply_to: reply_to}}
-    update_subscriptions_after_delivering_message(state, sid)
+    unless is_nil(state.receivers[sid]) do
+      send state.receivers[sid].recipient, {:msg, %{topic: topic, body: body, reply_to: reply_to}}
+      update_subscriptions_after_delivering_message(state, sid)
+    end
```
Comment (mtokioka) [inline lib/gnat.ex]: Sorry. I forget "else"

**Comment by mmmries** [comment]
Do you have an example of where this is useful to check? Does this just handle the race condition where we `unsub` from a topic, but we still have to process the last few messages for that `sid`?

**Comment by mtokioka** [comment]
This just handle the race condition.
Publish from another process sometime reach subscriber after ```unsub``` and cause reference of ```nil.recipient```.

**Comment by mtokioka** [comment]
I try to reproduce with test script but cannot do.

**Comment by mtokioka** [comment]
This code *sometimes* cause exceptions. (dirty...)
```
     ** (EXIT from #PID<0.164.0>) an exception was raised:
         ** (UndefinedFunctionError) function nil.recipient/0 is undefined or private
             nil.recipient()
             (gnat) lib/gnat.ex:253: Gnat.process_message/2
             (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
             (gnat) lib/gnat.ex:169: Gnat.handle_info/2
             (stdlib) gen_server.erl:601: :gen_server.try_dispatch/4
             (stdlib) gen_server.erl:667: :gen_server.handle_msg/5
             (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

```
  test "race condition" do
    topic = "test_race_condition"
    msg = "test msg"
    {:ok, gnat_sub} = Gnat.start_link()
    {:ok, gnat_pub} = Gnat.start_link()

    {:ok, subscription} = Gnat.sub(gnat_sub, self(), topic)
    Gnat.unsub(gnat_sub, subscription)
    {:ok, subscription} = Gnat.sub(gnat_sub, self(), topic)
    :ok = Gnat.pub(gnat_pub, topic, msg)
    Gnat.unsub(gnat_sub, subscription)
    {:ok, subscription} = Gnat.sub(gnat_sub, self(), topic)
    :ok = Gnat.pub(gnat_pub, topic, msg)
    Gnat.unsub(gnat_sub, subscription)
    {:ok, _subscription} = Gnat.sub(gnat_sub, self(), topic)
  end
```

**Comment by mmmries** [comment]
Released as part of `0.3.3`

**Pull request merged**